Tests/test_map.gd

```diff
@@ -2,6 +2,20 @@ extends GutTest ## Tests for MapManager, more will be added in the future
 
 var test_generator = load("res://Map/MapManager.gd")
 
+
+
 func test_map_gen():
-	var result = test_generator.create_map()
-	assert_not_null(result, "Expected to return generated class")
+	var test_width: Array[int] = [1,3,5,3,1]
```

Comment:
good way of testing it. Just to be exhaustive I think it should be good to test that the place where we expect a room is actually not null
Also from a code standpoint, maybe doing the following would be clearer:
```GDScript
var test_width: Array[int] = [1,3,5,3,1]
var test_map: MapBase = test_generator.create_map(test_width)
var expected_null_array: Array[Array] = [[0,0],[0,1],[0,3],[0,4],[1,0] ...]
for couple: Array[int] in expected_null_array:
    assert_null(test_map.rooms[couple[0]][couple[1]])
# you can another loop for the not null
```
need to check syntax since I just wrote it like that, but this allows to not add a new assert line every time you add more things (and the code is also shorter). One down side of that might if we have an error, it'll always be the same line. To prevent not knowing which position gives the error, a print that gives the current room would be good (so that if it fails, we know where we were)